packages/shared/src/GeolocationContext.js

```diff
@@ -1,39 +1,89 @@
 // @flow strict
 
 import * as React from 'react';
+import { Linking } from 'react-native';
+import { Alert } from '@kiwicom/mobile-localization';
 
 const defaultState = {
   lat: null,
   lng: null,
-  canGetUserLocation: true,
+  canGetUserLocation: false,
+  actions: { updateGeolocation: () => {} },
 };
 
 const { Consumer, Provider: ContextProvider } = React.createContext({
   ...defaultState,
 });
 
+type Coordinate = {|
+  +latitude: number,
+  +longitude: number,
+|};
+
 type Props = {|
-  children: React.Node,
+  +children: React.Node,
+  +shouldNotUpdateGeolocationComponentDidMount?: boolean,
 |};
 
 type State = {|
   lat: number | null,
   lng: number | null,
   canGetUserLocation: boolean,
+  actions: {|
+    +updateGeolocation: (
+      dealWithLocation?: (coordinate: Coordinate) => void,
+      onError?: () => void,
+    ) => void,
+  |},
 |};
 
 class Provider extends React.Component<Props, State> {
-  state = {
-    ...defaultState,
-  };
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      ...defaultState,
+      actions: {
+        updateGeolocation: this.updateGeolocation,
+      },
+    };
+  }
 
   componentDidMount = () => {
-    this.updateGeolocation();
+    this.updateGeolocation(() => {}, () => {});
   };
 
-  updateGeolocation = () => {
+  openSettings = () => {
+    Linking.openURL('app-settings:');
+  };
+
+  alertOpenSettings = () => {
+    Alert.translatedAlert(
+      undefined,
+      {
+        id: 'mmb.trip_services.transportation.map.current_position_alert',
+      },
+      [
+        {
+          text: { id: 'mmb.alert.button.ok' },
+          undefined,
+          style: 'default',
+        },
+        {
+          text: { id: 'mmb.alert.button.settings' },
+          onPress: this.openSettings,
+          style: 'default',
+        },
+      ],
+    );
+  };
+
+  updateGeolocation = (
+    dealWithLocation?: (coordinate: Coordinate) => void = () => {},
+    onError?: () => void = this.alertOpenSettings,
```

Comment:
just leave this as `onError?: () => void` and let the consumer decide what to do, so we don't have to pass in two empty functions every time